modules/analysis-common/src/main/java/org/opensearch/analysis/common/CommonAnalysisModulePlugin.java

```diff
@@ -343,6 +342,36 @@ public Map<String, AnalysisProvider<TokenFilterFactory>> getTokenFilters() {
         return filters;
     }
 
+    @Override
+    public Map<String, AnalysisProvider<TokenFilterFactory>> getTokenFilters(AnalysisModule analysisModule) {
```

Comment:
Rather than adding an overloaded implementation, can you just modify the existing `getTokenFilters` to take the `AnalysisModule`? We won't need the no-args version in this plugin.

Reply:
Just left the existing `getTokenFilters()` for backward compatibility, the  `getTokenFilters`  with `AnalysisModule` starts by invoking the default `getTokenFilters()`.

Reply:
Example https://github.com/search?q=repo%3Aopensearch-project%2FOpenSearch%20getTokenFilters()&type=code

Reply:
Oh -- I'm not suggesting getting rid of `getTokenFilters` in general. I'm just saying that for this specific plugin, we don't need the override.

With your change to `AnalysisModule`, it will always pass itself to `getTokenFilters(AnalysisModule)`. The default behavior for any `AnalysisPlugin` is to ignore that and delegate to the no-args `getTokenFilters()`. For *this* specific plugin, we do care about the `AnalysisModule`, so we can implement `getTokenFilters(AnalysisModule)` (and we don't need to implement `getTokenFilters()`).